pkg/controllers/management/gke/gke_cluster_handler.go

```diff
@@ -203,9 +203,9 @@ func (e *gkeOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			if err != nil {
```

Comment:
> I also changed the implementation for GKE. These changes will have to be backported to 2.5

Can you clarify that part, I see you changed the gke cluster handler but why does it need to be backported, just because its a bug ?

Reply:
Yes. We only want to set `PrivateRequiresTunnel` to a non-nil value if we know the answer for sure.

Reply:
Ok, so this might be a performance/overhead problem because we are tunneling when we don't need to be. Can you make a backport issue and put into 2.5.10 and link backport PR there.

Reply:
I believe this does the opposite. If `PrivateRequiresTunnel == nil`, then we don't tunnel. So this will protect against tunneling unless we are able to determine that we must do so.